Plugins/GeoModel/src/GeoModelDetectorElementITk.cpp

```diff
@@ -0,0 +1,58 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/GeoModel/GeoModelDetectorElementITk.hpp"
+
+#include "Acts/Surfaces/AnnulusBounds.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+
+namespace Acts {
+
+std::tuple<std::shared_ptr<Acts::GeoModelDetectorElementITk>,
+           std::shared_ptr<Acts::Surface>>
+Acts::GeoModelDetectorElementITk::convertFromGeomodel(
```

Comment:
Could this be a plain old constructor?

Reply:
I would like to have a function that makes a transform (detEl, surface) -> (detEl, surface), while in the end everything is correctly connected. This is not possible with a plain constructor.

Reply:
If you need the surface to be returned externally, that's true. But do you? It might make sense to assemble the pair at the call site. The detector element contains the surface anyway, so the pair return type is redundant in some sense.

Reply:
I don't see why it makes sense to assemble the pair on the call site (which might be on python).
The fact that the types are redundant in is true, but that is the current pattern in the plugin, and it would require additional bindings etc. to make it work on the python side. I wouldn't want to touch this in that PR. Can we just leave it like this for now?